src/OpenTelemetry.Instrumentation.AWSLambda/Implementation/AWSLambdaUtils.cs

```diff
@@ -40,7 +41,19 @@ public AWSLambdaUtils(AWSSemanticConventions semanticConventionBuilder)
 
     internal static ActivityContext GetXRayParentContext()
     {
-        var tracerHeaderValue = LambdaTraceProvider.CurrentTraceId;
+        string? tracerHeaderValue;
+        try
+        {
+            // Prefer the TraceProviderIsolated.CurrentTraceId over the environment variable to support
+            // AWS Lambda's Managed Instances feature. With Managed Instances there are multiple invocations
+            // per execution and environment variable is not set since there isn't a single value to set to.
+            tracerHeaderValue = TraceProviderIsolated.CurrentTraceId;
+        }
+        catch (TypeLoadException)
```

Comment:
Is there a way to cache which of the two approaches works for a given process? Otherwise I think we'd take a perf hit for each call from the try-catch?